dotnet/src/SemanticKernel.Skills/Skills.Web/WebFileDownloadSkill.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.SemanticKernel.Orchestration;
+using Microsoft.SemanticKernel.SkillDefinition;
+
+namespace Microsoft.SemanticKernel.Skills.Web;
+
+/// <summary>
+/// Skill to download web files.
+/// </summary>
+public class WebFileDownloadSkill : IDisposable
+{
+    /// <summary>
+    /// Parameter names.
+    /// <see cref="ContextVariables"/>
+    /// </summary>
+    public static class Parameters
+    {
+        /// <summary>
+        /// Where to save file.
+        /// </summary>
+        public const string FilePath = "filePath";
+    }
+
+    private readonly ILogger _logger;
+    private readonly HttpClientHandler _httpClientHandler;
+    private readonly HttpClient _httpClient;
+
+    /// <summary>
+    /// Constructor for WebFileDownloadSkill.
+    /// </summary>
+    /// <param name="logger">Optional logger.</param>
+    public WebFileDownloadSkill(ILogger<WebFileDownloadSkill>? logger = null)
+    {
+        this._logger = logger ?? NullLogger<WebFileDownloadSkill>.Instance;
+        this._httpClientHandler = new() { CheckCertificateRevocationList = true };
+        this._httpClient = new HttpClient(this._httpClientHandler);
+    }
+
+    /// <summary>
+    /// Downloads a file to local storage.
+    /// </summary>
+    /// <param name="uri">URI of file to download</param>
+    /// <param name="context">Semantic Kernel context</param>
+    /// <returns>Task.</returns>
+    /// <exception cref="KeyNotFoundException">Thrown when the location where to download the file is not provided</exception>
+    [SKFunction("Downloads a file to local storage")]
+    [SKFunctionInput(Description = "URL of file to download")]
+    [SKFunctionContextParameter(Name = Parameters.FilePath, Description = "Path where to save file locally")]
+    [SuppressMessage("Design", "CA1054:URI-like parameters should not be strings", Justification = "Semantic Kernel operates on strings")]
+    public async Task DownloadToFileAsync(string uri, SKContext context)
```

Comment:
Suggestion: it'd be a little cleaner to have `public async Task DownloadToFileAsync(string source, SKContext context)` - reduces format warning suppressions

Reply:
Done